feature/ICL-120 guides (#424)
energywebfoundation/iam-client-lib: docs/guides/organization.md

# Organization

[Organization Interface](../api/interfaces/cacheServerClient_cacheServerClient_types.IOrganization.md)

> **Organization** is basic type in EnergyWeb domains tree. Its main purpose is to be a host.

Reviewer: What do you mean by "be a host"?

Author: I tried to express that organization serves as container or root which other domains attached to. May be there is more appropriate term for this?

Reviewer: I think that either of the terms, "container" or "root", would be good, however how about the text below?
```suggestion
**Organization** is basic type in EnergyWeb domains tree. Its main purpose is to be a root that can group other domains, such as apps and roles.
```

All other domains such as applications and roles should have parent organization.

> If needed organization can include other sub-organizations.Organization is represented by

Reviewer: ```suggestion
If needed organization can include other sub-organizations. Organization is represented by
```

> [`IOrganization`](../api/interfaces/IOrganization.md) interface. Organization can

Reviewer: ```suggestion
[`IOrganization`](../api/interfaces/IOrganization.md) interface. Organizations can
```

> be created either as way to structure system or group correlated entities along with permissions hieararchy.

Reviewer: What do you mean by "structure system" here?

Organization owner as well as its components owners can be changed.
Example of organization

```

> {
>   "id": 544,
>   "name": "OKE",
>   "namespace": "OKE.iam.ewc",
>   "namehash": "0x01894757df8b51871036dd8042c86214cb286d7fb30f6ce868b6d337885ca1cb",
>   "owner": "0x5f7***",
>   "definition": {
>     "orgName": "OKE"
>   },
>   "apps": [
>     ...
>   ],
>   "subOrgs": [],
>   "roles": [
>     ...
>   ]
> }

Reviewer: Better to use the EnergyWeb org here as it is more professional
```suggestion
{
  "id": 24,
  "name": "energyweb",
  "namespace": "energyweb.iam.ewc",
  "namehash": "0x9c0d998fec800b14f2aa5b83643561cab9a7f52630b5c3889cf5e2a7a175d632",
  "owner": "0x627306090abaB3A6e1400e9345bC60c78a8BEf57",
  "definition": {
    "others": {
      "logoUrl": "https://stake.energyweb.org/assets/img/ew-flex-single-logo.png"
    },
    "logoUrl": "https://stake.energyweb.org/assets/img/ew-flex-single-logo.png",
    "orgName": "Energy Web Foundation",
    "websiteUrl": "https://energyweb.org",
    "description": "Energy Web (EW) is a global nonprofit organization accelerating a low-carbon, customer-centric electricity system by unleashing the potential of open-source, decentralized technologies. EW focuses on building core infrastructure and shared technology, speeding adoption of commercial solutions, and fostering a community of practice."
  },
  "subOrgs": []
}
```
